src/thunderbots/software/ai/navigator/obstacle/obstacle.cpp

```diff
@@ -0,0 +1,109 @@
+    Vector velocity_cushion_vector =
+        robot.velocity().norm(robot.velocity().len() * tick_length);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion);
+}
+
+Obstacle Obstacle::createRobotObstacleWithScalingParams(
+    const Robot& robot, bool enable_velocity_cushion, double radius_cushion_scaling,
+    double tick_length, double velocity_cushion_scaling)
+{
+    double radius_cushion = ROBOT_MAX_RADIUS_METERS * 4 / std::sqrt(3) * radius_cushion_scaling;
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector = robot.velocity().norm(
+        robot.velocity().len() * tick_length * velocity_cushion_scaling);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion && velocity_cushion_vector.len() > radius_cushion);
+}
+
+Obstacle Obstacle::createRobotObstacleWithBufferParams(
+    const Robot& robot, bool enable_velocity_cushion, double radius_cushion_buffer,
+    double tick_length, double velocity_cushion_buffer)
+{
+    double radius_cushion = ROBOT_MAX_RADIUS_METERS * 4 / std::sqrt(3) + radius_cushion_buffer;
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector = robot.velocity().norm(
+        robot.velocity().len() * tick_length + velocity_cushion_buffer);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion && velocity_cushion_vector.len() > radius_cushion);
+}
+
+
+Obstacle Obstacle::createRobotObstacleFromPositionAndRadiusAndVelocity(
+    Point position, double radius_cushion, Vector velocity_cushion_vector,
+    bool enable_velocity_cushion)
+{
+    if (enable_velocity_cushion)
+    {
+        Vector velocity_direction_norm_radius =
+            velocity_cushion_vector.norm(radius_cushion);
+        return Obstacle(Polygon(
+            {// left side of robot
+             position + velocity_direction_norm_radius.rotate(Angle::quarter()),
+             // back left of robot
+             position + velocity_direction_norm_radius.rotate(Angle::ofDegrees(150)),
+             // back right of robot
+             position + velocity_direction_norm_radius.rotate(Angle::ofDegrees(210)),
+             // right side of robot
+             position + velocity_direction_norm_radius.rotate(Angle::threeQuarter()),
+             // right side velocity cushions
+             position + velocity_direction_norm_radius.rotate(Angle::threeQuarter()) +
+                 velocity_cushion_vector,
+             // left side velocity cushions
+             position + velocity_direction_norm_radius.rotate(Angle::quarter()) +
```

Comment:
Maybe it makes more sense to create a new shape type of an circle elongated along the center. It could be a combination of 2 circles (at the ends) and a rectangle connecting them. The length of the rectangle would be proportional to the velocity in this case.

It's smart to use a polygon because it is a generic shape, but I wonder if it's worth it as all objects on the field are roughly circular I would think.

Thoughts anyone?

Reply:
With polygons we get a finite set of points (6) which might make other calculations easier. 
Current shape is a lopsided hexagon:
          x ---------------- x
     /                           |
x                               |
|                                |
x                               |
     \                           |
          x ---------------- x

Reply:
Sorry for not seeing this earlier. I definitely see where you're coming from on this on Evan, but I think for now just using a Polygon keeps things pretty simple, so I'm inclined to stick with that.